2996_reflection/reflection.md

```diff
@@ -7252,17 +7311,18 @@ concept reflection_range =
   same_as<remove_cvref_t<ranges::range_reference_t<R>>, info>;
 ```
 
-[1]{.pnum} *Constant When*: Let
-
 ```cpp
 template <reflection_range R = initializer_list<info>>
 consteval bool can_substitute(info templ, R&& arguments);
 ```
-[1]{.pnum} *Constant When*: `templ` represents a template and every reflection in `arguments` represents a construct usable as a template argument ([temp.arg]).
+[1]{.pnum} Let {`TArgs@~_N_~@`} be a sequence of template arguments corresponding to the reflections in `arguments` (call them {`R@~_N_~@`}) defined as follows:
+
+- [#.#]{.pnum} If `R@~_K_~@` represents an object or a value, then `TArgs@~_K_~@` is an expression whose result is the the represented object or value. The expression is an lvalue if and only if `R@~_K_~@` is an object and is a bit-field if and only if `R@~_K_~@` is a bit-field.
+- [#.#]{.pnum} Otherwise, `TArgs@~_K_~@` is the construct represented by `R@~_K_~@`.
 
-[#]{.pnum} Let `Z` be the template represented by `templ` and let `Args...` be the sequence of entities, values, and objects represented by the elements of `arguments`.
+[1]{.pnum} *Constant When*: `templ` represents a template and every reflection in the sequence {`TArgs@~_N_~@`} is a construct usable as a template argument ([temp.arg]).
 
-[#]{.pnum} *Returns*: `true` if `Z<Args...>` is a valid *template-id* ([temp.names]). Otherwise, `false`.
+[#]{.pnum} *Returns*: Letting `Z` be the template represented by `templ` and `TArgs...` be the sequence of constructs derived from `arguments` as described above, `true` if `Z<TArgs...>` is a valid `$template-id$` ([temp.names]). Otherwise, `false`.
```

Comment:
This seems a lot more complex, what's the benefit?

If we change anything can probably reduce to:

> [#]{.pnum} Let `Z` be the template represented by `templ` and let `Args...` be the sequence of constructs represented by the elements of `arguments`.

Reply:
Agreed, but the non-type template arguments in a _template-id_ aren't values or objects; they're expressions. We need to express that "transformation" from a value/object to an expression.